components/kyma-environment-broker/internal/event-hub/event_hub_step.go

```diff
@@ -0,0 +1,209 @@
+package event_hub
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/ptr"
+
+	"github.com/Azure/azure-sdk-for-go/services/eventhub/mgmt/2017-04-01/eventhub"
+	"github.com/kyma-incubator/compass/components/provisioner/pkg/gqlschema"
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/event-hub/azure"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+)
+
+const (
+	authorizationRuleName = "RootManageSharedAccessKey"
+
+	kafkaPort = 9093
+
+	k8sSecretNamespace                = "knative-eventing"
+	componentNameKnativeEventing      = "knative-eventing"
+	componentNameKnativeEventingKafka = "knative-eventing-kafka"
+	kafkaProvider                     = "azure"
+)
+
+// ensure the interface is implemented
+var _ process.Step = (*ProvisionAzureEventHubStep)(nil)
+
+type ProvisionAzureEventHubStep struct {
+	operationManager    *process.OperationManager
+	hyperscalerProvider azure.HyperscalerProvider
+	accountProvider     hyperscaler.AccountProvider
+	context             context.Context
+}
+
+func NewProvisionAzureEventHubStep(os storage.Operations, hyperscalerProvider azure.HyperscalerProvider, accountProvider hyperscaler.AccountProvider, ctx context.Context) *ProvisionAzureEventHubStep {
+	return &ProvisionAzureEventHubStep{
+		operationManager:    process.NewOperationManager(os),
+		accountProvider:     accountProvider,
+		context:             ctx,
+		hyperscalerProvider: hyperscalerProvider,
+	}
+}
+
+func (p *ProvisionAzureEventHubStep) Name() string {
+	return "Provision Azure Event Hubs"
+}
+
+func (p *ProvisionAzureEventHubStep) Run(operation internal.ProvisioningOperation,
+	log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+
+	hypType := hyperscaler.Azure
+
+	// parse provisioning parameters
+	pp, err := operation.GetProvisioningParameters()
+	if err != nil {
+		// if the parameters are incorrect, there is no reason to retry the operation
+		// a new request has to be issued by the user
+		log.Errorf("Aborting after failing to get valid operation provisioning parameters: %v", err)
```

Comment:
Good question!

The problem only appears if the provisioning parameters cannot be unmarshalled (e.g. invalid json). The request to the KymaEnvironmentBroker (KEB) is send via the UI later, so I hope the UI will display a reasonable error message. 

The step operation is async, so ideally the provisioning parameters (which are part of the HTTP PUT request), are validated before the async operation started and give the user a 4XX status code. But to be honest, I don't know if this is the case. 

Since the code `operation.GetProvisioningParameters()` is used heavily throughout KEB, I would leave it to the authors of KEB to implement fast user feedback :)